### PR TITLE
refactor(modules): use whoami crate to get username

### PR DESCRIPTION
**Pull request by bestgopher**
#### Description

1. get username by `whoami` crate, because the `USER` env may be changed.
2. fix some clippy warning
3. get hostname by `whoami`, and remove the `gethostname` crate from dependencies.

#### Motivation and Context

Closes #

#### How Has This Been Tested?

- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [ ] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

File: src/modules/username.rs [hunk 1]
```diff
@@ -2,22 +2,15 @@ use super::{Context, Module, ModuleConfig};
...
+#[cfg(not(test))]
+use whoami::username;
 
-#[cfg(not(target_os = "windows"))]
-const USERNAME_ENV_VAR: &str = "USER";
-
-#[cfg(target_os = "windows")]
-const USERNAME_ENV_VAR: &str = "USERNAME";
+#[cfg(test)]
+use tests::username;
```
Comment (davidkna) [inline src/modules/username.rs:9]: I would prefer handling this with a small wrapper function, and it seems like a good idea to keep around the possibility of overriding the username in the tests:
```rust
fn username() -> String {
    #[cfg(test)]
    if let Some(user) = context.get_env("USER") {
        return user;
    };
    whoami::username()
}

Comment (bestgopher) [reply src/modules/username.rs]: Thanks for your suggest. I have  optimized the codes, and review again please.

**Review by davidkna — APPROVED** [review]

**Comment by bestgopher** [comment]
@davidkna Hi. Is there anything else here that I need to fix?

**Comment by davidkna** [comment]
Thanks for the contribution @bestgopher!

**Pull request merged**